### PR TITLE
Fixes #72 Create a test library to be used in vignettes

### LINKED ISSUE #9: Remove 32 bit support
@PavelBal @Yuri05 @msevestre @Ian-Du 

We don't need this I suppose?

### LINKED ISSUE #53: `loadAssembly()` should return a success flag
Currently, it is not clear if loading an assembly was a success or not.

The C++ implementation of `rclr_load_assembly` should either return a flag (sucessfull/not) or throw an error/message.

### LINKED ISSUE #72: Create a test library to be used in vignettes
Can we create a separate library that can be shipped out with the package and used in documentation?

### LINKED ISSUE #79: Update JSON for Modern C++
Currently used version is 3.11.1, there is a release 3.11.3 available.

### PR DESCRIPTION
**Pull request by rwmcintosh**
Fixes #72 Create a test library to be used in vignettes
Fixes #9 Remove 32 bit support
Fixes #53 loadAssembly() should return a success flag
Fixes #79 json.hpp not required

### CODE REVIEW

**Comment by rwmcintosh** [comment]
Just a start, you can see the R code that I wrote to test :)

You can either add the C# code, or write the R code you want and I'll add the C# code .

Whatever you add, we'll just make it work in this PR and merge it.

File: tests/testthat/test-basic.R [hunk 1]
```diff
@@ -1,123 +1,130 @@
...
+  actual <- testObj$call("TestParams", "Hello, ", "World!", 1L, 2L, 3L, 6L, 5L, 4L)
+  expected <- "Hello, World!123654"
+  expect_equal(actual, expected = expected)
+  actual <- testObj$call("TestParams", "Hello, ", "World!", as.integer(1:6))
+  expected <- "Hello, World!123456"
+  expect_equal(actual, expected = expected)
+})
+
+test_that("Vignette examples work", {
+  loadAssembly("rSharp.Examples")
```
Comment (rwmcintosh) [inline tests/testthat/test-basic.R:12]: Not sure how to test for a value here. I just keep seeing a string in R

**Review by PavelBal — APPROVED** [review]

**Comment by PavelBal** [comment]
@rwmcintosh @Ian-Du  Please re-review and lets merge this.

**Comment by PavelBal** [comment]
After merging develop into this branch, I cannot build the dlls any more.

File: inst/extdata/RSharp.runtimeconfig.json
Comment (rwmcintosh) [inline inst/extdata/RSharp.runtimeconfig.json:1]: Do we need this?

**Pull request merged**